modules/gdscript/gdscript_parser.cpp

```diff
@@ -4459,6 +4472,120 @@ bool GDScriptParser::export_annotations(AnnotationNode *p_annotation, Node *p_ta
 			push_error(vformat(R"(Node export is only supported in Node-derived classes, but the current class inherits "%s".)", p_class->base_type.to_string()), p_annotation);
 			return false;
 		}
+
+		if (is_dict) {
```

Comment:
I thiiink this is fine. This bit was tricky because I had to parse key & value independently, so there was a fair bit of repeat code as a result. It does the same logic as the outer `if (is_array)` check, albeit resolved earlier.

Reply:
I think this is fine. But eventually we should extract the logic to avoid the repetition and make sure it's consistent.